src/handlers/Registrar.ts

```diff
@@ -1,51 +1,54 @@
-import { type Context, type Event } from "ponder:registry";
+import { type Context } from "ponder:registry";
 import { domains, registrations } from "ponder:schema";
 import { type Hex, namehash } from "viem";
 import { isLabelValid, makeSubnodeNamehash, tokenIdToLabel } from "../lib/ens-helpers";
-import { NsReturnType } from "../lib/plugins";
 import { upsertAccount, upsertRegistration } from "../lib/upserts";
 
-type NsType<T extends string> = NsReturnType<T, "/eth">;
-
 const GRACE_PERIOD_SECONDS = 7776000n; // 90 days in seconds
 
-export const makeRegistryHandlers = (baseName: `.${`${string}`}eth`) => {
-  const baseNameNode = namehash(baseName.slice(1));
+/**
+ * A factory function that returns Ponder indexing handlers for a specified index name/subname.
+ */
+export const makeRegistryHandlers = (indexedSubname: `${string}eth`) => {
+  const indexedSubnameNode = namehash(indexedSubname);
 
   async function setNamePreimage(context: Context, name: string, label: Hex, cost: bigint) {
     if (!isLabelValid(name)) return;
 
-    const node = makeSubnodeNamehash(baseNameNode, label);
+    const node = makeSubnodeNamehash(indexedSubnameNode, label);
     const domain = await context.db.find(domains, { id: node });
     if (!domain) throw new Error("domain expected");
 
     if (domain.labelName !== name) {
       await context.db
         .update(domains, { id: node })
-        .set({ labelName: name, name: `${name}${baseName}` });
+        .set({ labelName: name, name: `${name}${indexedSubname}` });
     }
 
     await context.db.update(registrations, { id: label }).set({ labelName: name, cost });
   }
 
   return {
-    get baseNameNode() {
-      return baseNameNode;
+    get indexedSubnameNode() {
```

Comment:
why provide this back, it will be known to the caller because they provided it as an input agument

Reply:
We need this value in the caller for some inline indexing handlers:
https://github.com/namehash/ens-multichain-indexer/blob/aad651cf532f1d89b214695994bb2d8e2574b7e3/src/ponder-ens-plugins/eth.base/handlers/Registrar.ts#L8-L17
https://github.com/namehash/ens-multichain-indexer/blob/aad651cf532f1d89b214695994bb2d8e2574b7e3/src/ponder-ens-plugins/eth.base/handlers/Registrar.ts#L31

Reply:
ah, hmm. i don't love this but it's ok for the moment 👍  we'll refactor this in v2